app/controllers/adhoq/queries_controller.rb

```diff
@@ -29,6 +29,12 @@ def update
       redirect_to @query
     end
 
+    def destroy
+      Adhoq::Query.find(params[:id]).destroy!
+      @queries = Adhoq::Query.recent_first
+      render 'adhoq/queries/index'
```

Comment:
Why don't you use redirect ?
Better to use the redirect, I think that it is good.